### ISSUE TITLE
Tabbed Interface cannot manage two different flagging directories

### ISSUE THREAD

**Issue by EgonFerri** [gradio-app/gradio#5692]
### Describe the bug

When using the `gr.TabbedInterface` to create multiple Gradio interfaces and flagging the results from both demos, all flagged results are saved to the directory specified for the second demo (`flagged_2`). This behavior is unexpected, as flagged results from each demo should be saved in their respective directories.

**Steps to Reproduce:**
1. Run the provided Python script to launch the Gradio app with two demos (`demo_1` and `demo_2`) within a tabbed interface.
2. Enter a name in each of the demos and click the submit button to trigger the `greet` function.
3. Flag the results from both demos.
4. Check the contents of the `flagged_1` and `flagged_2` directories.

**Expected Result:**
Flagged results from `demo_1` should be saved in the `flagged_1` directory, and flagged results from `demo_2` should be saved in the `flagged_2` directory.

**Actual Result:**
All flagged results are saved to the `flagged_2` directory, regardless of which demo they originated from. This behavior is not in line with the expected behavior, as it mixes the flagged results from both demos in the `flagged_2` directory.

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

```
import gradio as gr


def greet(name):
    return "Hello " + name + "!"


demo_1 = gr.Interface(fn=greet, inputs="text", outputs="text", flagging_dir='flagged_1')
demo_2 = gr.Interface(fn=greet, inputs="text", outputs="text", flagging_dir='flagged_2')

demo_tabbed = gr.TabbedInterface(
    [demo_1, demo_2],
    ["demo1", "demo2"],
    title="Tabbed demo")

if __name__ == "__main__":
    demo_tabbed.launch()
```

If you open the app, try both demos, and flag both results, everything is saved to flagged_2

### System Info

```shell
**Environment:**
- Operating System: mac os
- Python Version: 3.9.4
- Gradio Version: 3.44.4
```

### Severity

Blocking usage of gradio

**Comment by ahmedxomar101** [comment]
I am also facing the same issue. For all tabs, flagging is saved for the last passed flagging_dir.
I am using gradio==3.44.4 version.

**Comment by harry-urek** [comment]
hey I was trying to solve this issue 
Can someone verify my findings I'm very new to Open Source 😔

**This is what I learned -**

- It's not a problem related to flagging_dir pointing towards a global variable
- In `Interface.py` `flagging_callback : FlaggingCallback = CSVLogger()` 
-  and in CSVLogger has classmethod called ```
setup(
        self,
        components: list[IOComponent],
        flagging_dir: str | Path,
    ):
 self.components = components
        self.flagging_dir = flagging_dir
        os.makedirs(flagging_dir, exist_ok=True)
```
- and the flag creation is done by another function called flag
 all of its dir paths seems to be pointing towards flagging_dir passed in setup() function 
- But the init function of CSVLogger is -
```
def __init__(self):
        `pass
````
Since `setup() ` is called during creation of the class instance but `flag()` function is called when flag is pressed and  `__init__() ` function is empty for CSVLogger file it takes the last created `flagging_dir` value since the class doesnt have a class_global value for `flagging_dir` it takes the last `flaggging_dir` value from when the setup was executed for demo_2 interface


I'm very new to the Open Source community so, I'm not getting any idea how to solve it properly one idea I had was to move the setup function to init() but I dont think thats a good idea

also I have little doubt I'm currently running the the code inside Dev Conntainer in VsCode if I make changes in Dev Container files will gradio automatically updates its running package or i would have to restart my dev container

Can

**Comment by EgonFerri** [comment]
I tried to solve the issue with a pull request. 
The issue was correctly described by @harry-urek, but I think his proposed solution couldn't work since CSVLogger does not have multiple instances for multiple flagging dirs, so moving setup to init wouldn't change anything. The setup runs correctly when each interface is instantiated, but flag() refers to the last self.flagging_dirs. I tried to solve it by adding a "flagging_dir" parameter to flag().
@abidlabs let me know what you think (:
@abidlabs let me know what you think (:
